### DISCUSSION TITLE
Slow installation of modules on Windows

### DISCUSSION THREAD

**Discussion by sewbacca** [keplerproject/luarocks#1562] [Ideas]
Through profiling I found out, that around half the time, luarocks is busy setting file permissions on Windows, when installing Lua only modules. If calls to `icacls` (in [src/luarocks/fs/win32/tools.lua:280-295](https://github.com/luarocks/luarocks/blob/89daf1f1588189f0fedc0b050408d785729a6833/src/luarocks/fs/win32/tools.lua#L280-L295)) are disabled, then there is a 2x speedup during installation, which is comparable to the time I've measured on Linux. Is there a reason / need for setting file permissions on Windows?

**Comment by hishamhm** [comment]
The intention was to add the same support for "multi-user system trees" and "single-user user trees" that is available on Unix. 

I was unaware of this slowdown. It could be a good idea to detect that the "user" tree is in use and skip the permission-setting on Windows, or make skipping it into a config option. PRs welcome!
